server_addon/houdini/client/ayon_houdini/plugins/load/load_hda.py

```diff
@@ -28,14 +28,17 @@ def load(self, context, name=None, namespace=None, data=None):
         # Get the root node
         obj = hou.node("/obj")
 
-        # Create a unique name
-        counter = 1
         namespace = namespace or context["folder"]["name"]
-        formatted = "{}_{}".format(namespace, name) if namespace else name
-        node_name = "{0}_{1:03d}".format(formatted, counter)
+        node_name = "{}_{}".format(namespace, name) if namespace else name
 
         hou.hda.installFile(file_path)
-        hda_node = obj.createNode(name, node_name)
+
+        # Get the type name from the HDA definition.
+        hda_defs = hou.hda.definitionsInFile(file_path)
+        for hda_def in hda_defs:
+            type_name = hda_def.nodeTypeName()
```

Comment:
Should this `break` on first iteration, or maybe take it from the last entry or?

Reply:
It's just from a coding perspective it's clearer what the intentions are. If you're always expecting one for example you could also do:
```python
type_name = hda_defs[0].nodeTypeName()
```
And then it's clear you're just taking the first one.

Reply:
what about ? 
```python
        # Get the type name from the HDA definition.
        hda_defs = hou.hda.definitionsInFile(file_path)
        if hda_defs:
          type_name = hda_defs[0].nodeTypeName()
```

Reply:
You'd still hit an error lower down on `type_name` being undefined?
I'd thus reverse this logic:

```python
        # Get the type name from the HDA definition.
        hda_defs = hou.hda.definitionsInFile(file_path)
        if not hda_defs:
            raise RuntimeError(f"No HDA definitions found in file: {file_path}")
        type_name = hda_defs[0].nodeTypeName()
```
Or, if there are always `hda_defs` expected in the returned value then your if statement would be redundant and I'd remove it.

Reply:
cool, let me push it.